### PR TITLE
Create an instance of  ProviderClient defined in golangsdk

### PR DESCRIPTION
**Pull request by zengchen1024**
The new ProviderClient will be used to send/receive messages to opentelekom cloud

This pr is based on #9

### CODE REVIEW

**Comment by zengchen1024** [comment]
@cgriggs01 Could you please merge this pr? Thanks very much!

**Comment by freesky-edward** [comment]
@paultyng @cgriggs01 could you please take a look at this PR. which proposes to add a new golangsdk client into config.go. the reason of adding this repository is to separate gophercloud as it was. so that it can have a better dependency management.

File: opentelekomcloud/config.go [hunk 1]
```diff
@@ -198,6 +211,90 @@ func (c *Config) LoadAndValidate() error {
...
+		UserID:           c.UserID,
+	}
+
+	client, err := huaweisdk.NewClient(ao.IdentityEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// Set UserAgent
+	client.UserAgent.Prepend(terraform.UserAgentString())
```
Comment (paultyng) [inline opentelekomcloud/config.go:207]: Just FYI this method is moving to [`httpclient.UserAgentString()`](https://github.com/hashicorp/terraform/blob/master/httpclient/useragent.go) but its also worth mentioning that currently this only shares the vendored TF version, not the CLI version used to invoke the provider, although we are looking to address that.

File: opentelekomcloud/config.go [hunk 2]
```diff
@@ -198,6 +211,90 @@ func (c *Config) LoadAndValidate() error {
...
+		caCertPool.AppendCertsFromPEM([]byte(caCert))
+		config.RootCAs = caCertPool
+	}
+
+	if c.Insecure {
+		config.InsecureSkipVerify = true
+	}
+
+	if c.ClientCertFile != "" && c.ClientKeyFile != "" {
+		clientCert, _, err := pathorcontents.Read(c.ClientCertFile)
```
Comment (paultyng) [inline opentelekomcloud/config.go]: You may want to move some of this common initialization in to a shared func. There is a good amount of repetition between the two initialization functions. The azurerm provider [has an example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/config.go) of how multiple client initialization has had a lot of the repetition cleaned up.

File: opentelekomcloud/config.go [hunk 3]
```diff
@@ -8,6 +8,8 @@ import (
 	"net/http"
 	"os"
 
+	"strings"
```
Comment (paultyng) [inline opentelekomcloud/config.go]: Got some unnecessary whitespace between `os` and `strings`

**Review by paultyng — CHANGES_REQUESTED** [review]

File: opentelekomcloud/config.go (hunk 1 above)
Comment (paultyng) [reply opentelekomcloud/config.go:207]: I don't think you have to change this unless your vendored TF has the new method.

File: opentelekomcloud/config.go (hunk 3 above)
Comment (zengchen1024) [reply opentelekomcloud/config.go]: Done

File: opentelekomcloud/config.go (hunk 2 above)
Comment (zengchen1024) [reply opentelekomcloud/config.go]: Good suggestion! Thanks!

**Comment by zengchen1024** [comment]
@paultyng Thanks very much for your comments! Could you take a look again. Thanks!

**Review by paultyng — APPROVED** [review]
This looks ok to me now, but could we add some acceptance testing? Or is it indirectly tested by other acceptance tests?

**Comment by zengchen1024** [comment]
@paultyng This pr is tested indirectly by this pr #11 whose acceptance tests are all passed.
